check_diff/src/lib.rs

```diff
@@ -56,3 +128,83 @@ pub fn change_directory_to_path(dest: &Path) -> io::Result<()> {
     );
     return Ok(());
 }
+
+// Compiles and produces two rustfmt binaries.
+// One for the current master, and another for the feature branch
+// Parameters:
+// dest: Directory where rustfmt will be cloned
+pub fn compile_rustfmt(
+    dest: &Path,
+    remote_repo_url: String,
+    feature_branch: String,
+    commit_hash: Option<String>,
+) -> Result<Command, CheckDiffError> {
```

Comment:
I think it would be best to keep the logic in `compile_rustfmt` focused on compiling a single binary. We can add a higher level function named `compile_rustfmt_binaries` (or something similar), which compiles both binaries and coordinates performing all the git actions.

I've also been thinking about this, and I'm not sure if [`Commad` ](https://doc.rust-lang.org/std/process/struct.Command.html#) is the best return type. Originally I thought it could be reused, but I don't think that anymore. Command doesn't give us a way to clone or copy it. It might be best for `compile_rustfmt` to return a closure that we can use to call the appropriate binary.

The `compile_rustfmt` function signature might look like this:
```rust
fn compile_rustfmt(/* other args */) -> impl Fn(&[&str]) -> std::io::Result<std::process::Output>;
```

Reply:
This sounds really cool actually. So the end goal is to use the binaries produced here elsewhere in the `check_diff` crate? Then actually isnt a string representation sufficient (ie a path to the binary)?

Reply:
Yeah, this could totally work if we use the path to the binary, though we'd also need to keep track of the `LD_LIBRARY_PATH` for that binary, and there would also be some duplicated boilerplate when calling the binary. That's why I suggested we return a closure that sets up and calls the binary.